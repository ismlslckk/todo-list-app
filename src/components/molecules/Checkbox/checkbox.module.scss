.round {
    position: relative;
}

.round label {
    background-color: #fff;
    border: 1px solid #ccc;
    border-radius: 50%;
    cursor: pointer;
    height: 24px;
    left: 0;
    position: absolute;
    top: 0;
    width: 24px;
}

.round label::after {
    border: 2px solid #fff;
    border-top: none;
    border-right: none;
    content: "";
    height: 6px;
    left: 7px;
    opacity: 0;
    position: absolute;
    top: 6px;
    transform: rotate(-45deg);
    width: 8px;
}

.round input[type="checkbox"] {
    visibility: hidden;
}

.round input[type="checkbox"]:checked+label {
    background-color: #6643b5;
    border-color: gray;
}

.round input[type="checkbox"]:checked+label::after {
    opacity: 1;
}