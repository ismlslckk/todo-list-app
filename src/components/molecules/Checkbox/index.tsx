import styles from './checkbox.module.scss';

const Checkbox = ({ ...props }) => (

  <div className={styles.round}>
    <input
      type="checkbox"
      {...props}
    />
    <label htmlFor={props.id} />
  </div>

);

export default Checkbox;
