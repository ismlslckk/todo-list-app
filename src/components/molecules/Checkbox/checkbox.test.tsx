import { render } from '@testing-library/react';
import reactTestRenderer from 'react-test-renderer';
import Checkbox from '.';

describe('checkbox component tests', () => {
  test('expect type should be checkbox', async () => {
    const checkbox:any = render(<Checkbox />);

    expect(checkbox.container.firstChild.firstChild).toHaveProperty('type', 'checkbox');
  });

  test('expect checkbox should be readOnly', async () => {
    const checkbox:any = render(<Checkbox readOnly />);

    expect(checkbox.container.firstChild.firstChild).toHaveProperty('readOnly');
  });

  test('expect checkbox should be disable', async () => {
    const checkbox:any = render(<Checkbox disabled />);

    expect(checkbox.container.firstChild.firstChild).toHaveProperty('disabled');
  });

  it('renders correctly', () => {
    const tree = reactTestRenderer
      .create(<Checkbox />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
