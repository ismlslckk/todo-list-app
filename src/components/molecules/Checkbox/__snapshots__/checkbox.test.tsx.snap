// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox component tests renders correctly 1`] = `
<div
  className="round"
>
  <input
    className="form-control"
    id="37aca860-ea4e-4ebe-bfad-b4f4c2633f6d"
    type="checkbox"
  />
  <label
    htmlFor="37aca860-ea4e-4ebe-bfad-b4f4c2633f6d"
  />
</div>
`;
